src/app/dev/DevToys.Tools/Tools/Converters/JsonTable/JsonTableConverterGuiTool.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System.Data;
+using DevToys.Tools.Tools.Converters.JsonTable;
+using Microsoft.Extensions.Logging;
+using static DevToys.Tools.Helpers.JsonTableHelper;
+
+namespace DevToys.Tools.Tools.Converters.JsonYaml;
+
+[Export(typeof(IGuiTool))]
+[Name("JsonTableConverter")]
+[ToolDisplayInformation(
+    IconFontName = "DevToys-Tools-Icons",
+    IconGlyph = '\u0109',
+    GroupName = PredefinedCommonToolGroupNames.Converters,
+    ResourceManagerAssemblyIdentifier = nameof(DevToysToolsResourceManagerAssemblyIdentifier),
+    ResourceManagerBaseName = "DevToys.Tools.Tools.Converters.JsonTable.JsonTableConverter",
+    ShortDisplayTitleResourceName = nameof(JsonTableConverter.ShortDisplayTitle),
+    LongDisplayTitleResourceName = nameof(JsonTableConverter.LongDisplayTitle),
+    DescriptionResourceName = nameof(JsonTableConverter.Description),
+    AccessibleNameResourceName = nameof(JsonTableConverter.AccessibleName),
+    SearchKeywordsResourceName = nameof(JsonTableConverter.SearchKeywords))]
+[AcceptedDataTypeName(PredefinedCommonDataTypeNames.Json)]
```

Comment:
I agree with you idea of only detection JSON arrays. You can do it by creating a new data type detector, like this one:
https://github.com/veler/DevToys/blob/e7ecbbb65914516fe83f829e8e6614ab79b91889/src/app/dev/DevToys.Tools/SmartDetection/JsonDataTypeDetector.cs#L6-L38

Make sure that your new data type detector has "Json" as base type name. For example:
```
[Export(typeof(IDataTypeDetector))]
[DataTypeName("JsonArray", baseName: PredefinedCommonDataTypeNames.Json)]
```

This will ensure that your JsonArray detector only runs if the `JsonDataTypeDetector` detected a valid JSON. In your detector, you will be able to safely assume that the `data` parameter is a well formatted JSON, since it has already been detected previously.
As a consequence on the UX, when your JsonArray detector finds a data it accepts, then only tools that support JsonArray will be recommended to the user.

Reply:
Cheers for the heads up about baseName.

Reply:
The detector looks great! It's not correctly consumed in the tool though. On line `132`, it should be:
```csharp
        if (dataTypeName == PredefinedCommonDataTypeNames.JsonArray &&
            parsedData is string jsonStrongTypedParsedData)
        {
            _inputTextArea.Text(jsonStrongTypedParsedData);
        }
```

Reply:
Oops, thanks!